src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/migration/Cas3UpdateApplicationOffenderNameJob.kt

```diff
@@ -0,0 +1,119 @@
+import java.util.UUID
+import java.util.stream.Collectors
+import javax.persistence.EntityManager
+
+class Cas3UpdateApplicationOffenderNameJob(
+  private val applicationRepository: UpdateTemporaryApplicationOffenderNameRepository,
+  private val offenderService: OffenderService,
+  private val userService: UserService,
+  private val entityManager: EntityManager,
+  private val pageSize: Int,
+  private val transactionTemplate: TransactionTemplate,
+) : MigrationJob() {
+  private val log = LoggerFactory.getLogger(this::class.java)
+  override val shouldRunInTransaction = false
+
+  @SuppressWarnings("MagicNumber", "TooGenericExceptionCaught")
+  override fun process() {
+    var page = 1
+    var hasNext = true
+    var slice: Slice<ApplicationCrn>
+    var offendersCrn = setOf<String>()
+
+    try {
+      val user = userService.getUserForRequest()
+
+      while (hasNext) {
+        log.info("Getting page $page for max page size $pageSize")
+        slice = applicationRepository.getCrnForTemporaryAccommodationApplication(PageRequest.of(0, pageSize))
+
+        offendersCrn = slice.content.map { it.getCrn() }.toSet()
+
+        log.info("Updating offenders name with crn ${offendersCrn.joinToString { "," }}")
+
+        val personInfos = splitAndRetrievePersonInfo(offendersCrn, user.deliusUsername)
+
+        slice.content.map {
+          val personInfo = personInfos[it.getCrn()] ?: PersonSummaryInfoResult.Unknown(it.getCrn())
+          val applicationId = it.getId()
+
+          transactionTemplate.executeWithoutResult {
+            try {
+              val offenderName = when (personInfo) {
+                is PersonSummaryInfoResult.Success.Full -> "${personInfo.summary.name.forename} ${personInfo.summary.name.surname}"
+                is PersonSummaryInfoResult.NotFound, is PersonSummaryInfoResult.Unknown -> throw NotFoundProblem(
+                  personInfo.crn,
+                  "Offender",
+                )
+                is PersonSummaryInfoResult.Success.Restricted -> throw ForbiddenProblem()
+              }
+
+              applicationRepository.updateName(applicationId, offenderName)
+            } catch (e: IllegalStateException) {
+              log.error(e.message)
+            }
+          }
+        }
+
+        entityManager.clear()
+        hasNext = slice.hasNext()
+        page += 1
+      }
+    } catch (exception: Exception) {
+      log.error("Unable to update offenders name with crn ${offendersCrn.joinToString { "," }}", exception)
+    }
+  }
+
+  private fun splitAndRetrievePersonInfo(crns: Set<String>, deliusUsername: String): Map<String, PersonSummaryInfoResult> {
+    val crnMap = ListUtils.partition(crns.toList(), pageSize)
+      .stream().map { crns ->
+        offenderService.getOffenderSummariesByCrns(crns.toSet(), deliusUsername).associateBy { it.crn }
+      }.collect(Collectors.toList())
+
+    return crnMap.flatMap { it.toList() }.toMap()
+  }
+}
+
+@Repository
```

Comment:
we can avoid the below additional repository and update query

Reply:
Replaced with application repository